openfe/setup/network_planning.py

```diff
@@ -1,52 +1,82 @@
+# This code is part of OpenFE and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/openfe
 import math
+from typing import Iterable
 
+from openfe.setup import Network, Molecule, AtomMapper
 
-class Network:
-    @classmethod
-    def from_lomap(self, lomap_file):
-        pass
 
-
-def generate_radial_graph(ligands, central_ligand, mappers, extra_scorers=None):
+def generate_radial_network(ligands: Iterable[Molecule],
+                            central_ligand: Molecule,
+                            mappers: Iterable[AtomMapper], scorer=None):
     """Radial Network generator
 
+    Also known as hub and spoke or star-map, this plans a Network where
+    all ligands are connected via a central ligand.
+
     Parameters
     ----------
-    ligands : list of rdkit Molecules
+    ligands : iterable of Molecules
       the ligands to arrange around the central ligand
-    central_ligand : rdkit Molecule
+    central_ligand : Molecule
       the ligand to use as the hub/central ligand
     mappers : iterable of AtomMappers
       mappers to use, at least 1 required
-    extra_scorers : iterable of Scorers, optional
-      extra ways to assign scores
+    scorer : scoring function, optional
+      a callable which returns a float for any AtomMapping.  Used to assign
+      scores to potential mappings, higher scores indicate worse mappings.
+
+    Raises
+    ------
+    ValueError
+      if no mapping between the central ligand and any other ligand can be
+      found
 
     Returns
     -------
     network : Network
-      will have an edge between each ligand and the central ligand, with the mapping
-      being the best possible mapping found using the supplied atom mappers.
+      will have an edge between each ligand and the central ligand, with the
+      mapping being the best possible mapping found using the supplied atom
+      mappers.
+      If no scorer is supplied, the first mapping provided by the iterable
+      of mappers will be used.
     """
-    n = Network()
+    edges = []
 
-    for l in ligands:
-        best_score = inf
+    for ligand in ligands:
+        best_score = math.inf
         best_mapping = None
 
         for mapper in mappers:
-            for mapping in mapper.suggest_mappings(central_ligand, l):
-                score = mapper.score_mapping(mapping)
+            for mapping in mapper.suggest_mappings(central_ligand, ligand):
+                if not scorer:
+                    best_mapping = mapping
+                    break
```

Comment:
The behaviour is kinda odd here and it kinda clashes a bit with how it's described in the docstring since you'll essentially default to taking the first suggested mapping of the last mapper.